[TNT-190] 트레이니 연결 요청 팝업 기능 구현

SeonJeongk opened the pull request:
## 📝 작업 내용

- Closes #53 

- 홈 화면 진입 시 연결 요청 팝업이 자동으로 표시되도록 구현했습니다.
    - 회원 조회 API를 호출하여 isConnected 값을 확인하고, 
    `연결되지 않은 상태`이고 `3일 동안 보지 않기`로 설정한 시간이 지났다면
    **다이얼로그를 표시**하도록 처리했습니다.

- 3일 동안 보지 않기 기능을 구현했습니다.
    - 사용자가 `3일 동안 보지 않기` 를 선택하면 다이얼로그가 닫히는 순간의 LocalDateTime을
    `ConnectLocalDataSource`에 저장하도록 구현했습니다.
    - 저장된 값과 현재 시간을 비교하여 3일(72시간)이 지나면 다시 다이얼로그를 띄우도록 처리했습니다.
    - 저장된 시간은 로그아웃 혹은 탈퇴 시 자동으로 초기화하도록 처리했습니다.

- 회원가입 및 프로필 사진과 관련된 수정 사항 포함되어 있습니다

## 📸 실행 화면
### 트레이너 연결 이후 팝업 노출
https://github.com/user-attachments/assets/dbc0152f-3b77-4c45-bbbf-361c4e6db898

### 3일동안 보지 않기 동작
https://github.com/user-attachments/assets/bd64ce58-aa3c-4f53-8684-a98b981ab32e





## 🙆🏻 리뷰 요청 사항
없습니다.. 영상에 보이는 연결 완료 화면 이미지 오류는 해결했습니다!
<img src="https://github.com/user-attachments/assets/0e21f645-26d1-4565-a30d-a97904154fa7" width=200/>


## 👀 레퍼런스

SeonJeongk committed:
[TNT-190] design: TnTCheckToggleDialog 텍스트 클릭 추가

SeonJeongk committed:
[TNT-190] feat: 트레이니 홈화면 연결 dialog 띄우기 구현

- '3일간 보지 않기' 기능 구현 필요

SeonJeongk committed:
[TNT-190] feat: 3일간 보지 않기 설정한 날짜 dataStore에 저장

SeonJeongk committed:
[TNT-190] fix: rebase 이후 오류 수정

SeonJeongk committed:
[TNT-000] refactor: 회원가입 이름 입력 에러 문구 수정 반영

SeonJeongk committed:
[TNT-190] refactor: uiResource -> coreR로 수정

SeonJeongk committed:
[TNT-000] refactor: 회원가입 달력 초기 노출 날짜 수정

SeonJeongk committed:
[TNT-000] refactor: viewModel에서 context 사용하지 않도록 개선

SeonJeongk committed:
[TNT-000] refactor: 회원가입 문구 수정

SeonJeongk committed:
[TNT-190] feat: isConnected 설정 추가 및 datasource 역할에 맞게 이름 변경

SeonJeongk committed:
[TNT-190] feat: isConnected 저장 및 삭제 구현

SeonJeongk committed:
[TNT-190] feat: ConnectRepository에 localDatasource 추가

SeonJeongk committed:
[TNT-000] fix: 완료 회차 입력 조건 및 회원 기본 프로필 사진 수정

- 현재 완료된 회차에 0 입력 가능하도록 조건 수정

SeonJeongk committed:
[TNT-190] refactor: 연결 다이얼로그 뒤로가기 막기

SeonJeongk committed:
[TNT-190] fix: 연결여부 저장하지 않도록 수정

SeonJeongk committed:
[TNT-190] feat: 트레이니 연결 촉구 Dialog 기능 구현

SeonJeongk committed:
[TNT-000] fix: 에러 이미지 추가

SeonJeongk committed:
[TNT-190] fix: 네트워크 에러 핸들링

SeonJeongk committed:
[TNT-190] refactor: 네이밍 수정

SeonJeongk committed:
[TNT-190] fix: onEach 삭제 및 네이밍 수정

hoyahozz commented on data/storage/src/main/java/co/kr/data/storage/source/ConnectLocalDataSource.kt:
```diff
@@ -0,0 +1,37 @@
+class ConnectLocalDataSource @Inject constructor(
+    @Named(CONNECT_STORAGE_NAME) private val connectPreferences: DataStore<Preferences>,
+) {
+    val homeDialogHiddenDate: Flow<String?> = connectPreferences.data.map { preferences ->
```
변수명이 UI와 의존성을 지니지 않도록

`ExplicitDeniedConnectDate` 와 같은 방식으로 네이밍하는게 좋을 것 같아요!

SeonJeongk replied:
https://github.com/YAPP-Github/TnT-Android/pull/110/commits/82412633b3a6844fe29bf82caf29a624acac1198

앗 넵 수정했습니다!

hoyahozz commented on feature/trainee/home/src/main/java/co/kr/tnt/trainee/home/TraineeHomeViewModel.kt:
```diff
@@ -96,14 +104,76 @@ internal class TraineeHomeViewModel @Inject constructor(
+
+            val lastHiddenDate = connectRepository.getHomeDialogHiddenDate().firstOrNull()
+            val isHidden = lastHiddenDate != null &&
+                Duration.between(lastHiddenDate, currentDateTime).toHours() < 72
```
`72` 상수 선언해두면 좋을 것 같아요 !_!

SeonJeongk replied:
https://github.com/YAPP-Github/TnT-Android/pull/110/commits/c7e63ad65a29b9693fcbffc1a3864ad496a0dd88

수정했습니다!

hoyahozz commented on feature/trainer/connect/src/main/java/co/kr/tnt/trainer/connect/TrainerConnectCompletePage.kt:
```diff
@@ -99,18 +107,25 @@ internal fun TrainerConnectCompletePage(
 @Composable
 private fun ProfileSection(
     profile: User,
+    context: Context,
```
추후에 `painter` 받아서 사용하는 쪽에서 `placeholder` 설정하는 방식으로 만들면 좋을 것 같아요

그럼 `context` 도 안받아도 될듯요!

SeonJeongk replied:
음! 나중에 수정해두겠습니다

hoyahozz reviewed (approved):
![image](https://github.com/user-attachments/assets/2d6287d0-66ec-4a79-b356-e27535a9434f)

SeonJeongk committed:
[TNT-190] refactor: 네이밍 변경

SeonJeongk committed:
[TNT-190] refactor: 시간 상수 선언

SeonJeongk committed:
[TNT-190] Merge branch 'develop' into feature/TNT-190

# Conflicts:
#	feature/trainee/signup/src/main/java/co/kr/tnt/trainee/signup/TraineeSignUpViewModel.kt
#	feature/trainer/signup/src/main/java/co/kr/tnt/trainer/signup/TrainerSignUpViewModel.kt
